CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(overlap_checker LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 11)
+
+include(CheckSymbolExists)
+check_symbol_exists(argp_parse "argp.h" HAVE_ARGP)
+if(NOT (HAVE_ARGP))
+  message(SEND_ERROR "GNU argp is required for command line parsing")
+endif()
+
+# OpenCascade stuff
+find_path(OCC_INC "Standard_Version.hxx" HINTS "/usr/include/opencascade" REQUIRED)
+include_directories(SYSTEM ${OCC_INC})
+link_libraries(
+  TKernel
+  TKBO TKBRep TKPrim
+  TKG3d TKLCAF TKMath
+  TKTopAlgo TKXCAF TKSTEP
+  TKXSBase TKXDESTEP TKShHealing)
+
+# pull in our external header-only libraries
+include_directories(SYSTEM "include")
+
+find_package(Catch2 REQUIRED)
+add_executable(test_runner src/test_runner.cpp src/document.cpp src/utils.cpp)
```

Comment:
Might be worth having this main cmakefile call a subsequent cmakefile in the src directory which is closer to the tools?

Reply:
Lets also submodule catch2 for simplicity

Reply:
the file is only ~30 lines long, not sure if it's worth splitting up is it?  if it was hundreds of lines I could see an argument, but I'm not sure how another small file would improve clarity

Reply:
more for stylistic and good practise reasons, best to have a root level cmake file, which calls those from higher directories